### PR TITLE
Add Tab Layout for User Profile Screen

### LINKED ISSUE #374: Add different tabs in user profile screen
This doesn't include reworking/adding the content of each tab. It just includes the overrall layout
- Tab for Favorite Parkings
- Tab for My Reviews
- Tab for My Images

### LINKED ISSUE #375: Rework review cards in profile screen
- Reuse Review Card from All Reviews, displaying Parking title rather than Author
- Clicking the card takes you on the "Details Screen" of the parking

### PR DESCRIPTION
**Pull request by Armetiis**
This PR introduces a new tab layout for the user profile screen, improving the organization and display of favorite parkings and user reviews. The changes enhance the user experience by providing a more structured and navigable interface.

image
image

### Key Changes:
1. Tab Layout Implementation:
   - Added a new `TabLayout` composable to handle the display of "Favorite Parkings" and "My Reviews" sections.
   - Implemented tab selection logic using `rememberSaveable` to persist the selected tab across recompositions.

2. Favorite Parkings Section:
   - Moved the favorite parkings display into its own tab.
   - Updated the layout to use a vertical `LazyColumn` instead of a horizontal `LazyRow` for better space utilization.

3. User Reviews Section:
   - Replaced the custom review card with the `ReviewCard` component from the All Reviews screen.
   - Implemented navigation to the parking details screen when a review card is clicked.
   
### Future Improvements:
1. Parking Cards Redesign:
   - The cards for each parking will be reworked in the future, using a design similar to (but not identical to) the cards in the ListScreen. This will be done by myself in a subsequent PR

2. Additional Tabs:
   - Plans to add new tabs for:
     a. Images posted by the user (@LouisJamesVasseur)
     b. User's parking contributions (maybe)

### Notes:
- The implementation ensures that only existing parkings are displayed in the reviews section, preventing potential null pointer exceptions in the case of a desynchronisation in the database
- The use of `rememberSaveable` for tab selection ensures a consistent user experience across recompositions.
- Small change was made in the viewmodel to ensure that the state is updated upon update (like/dislikes are updates)

#### Review Navigation
Clicking a review in the user profile now leads to the parking details screen instead of the All Reviews screen. This choice was made because from the parking details, users can navigate to the All Reviews screen, but not vice versa. This ensures users can easily remind themselves of the parking context for their review, which wouldn't be possible if they could only access the All Reviews screen.

#### Not implementing Delete and Modify Options
Immediate options for deleting or modifying reviews have been removed from the user profile screen's review cards. This encourages users to revisit the parking details and review context before making changes, promoting more informed decisions about their reviews.

### Coverage
image
-- -
- Closes #374
- Closes #375

### CODE REVIEW

File: app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt [hunk 1]
```diff
@@ -183,49 +179,71 @@ fun ViewProfileScreen(
...
         style = MaterialTheme.typography.bodyMedium,
-        modifier = Modifier.testTag("NoFavoritesMessage"))
+        modifier = Modifier.testTag("NoFavoritesMessage").padding(16.dp))
   } else {
-    LazyRow(
-        modifier = Modifier.fillMaxWidth().testTag("FavoriteParkingList"),
-        horizontalArrangement = Arrangement.spacedBy(8.dp)) {
+    LazyColumn(
+        modifier = Modifier.fillMaxWidth().padding(16.dp).testTag("FavoriteParkingList"),
+        verticalArrangement = Arrangement.spacedBy(8.dp)) {
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt:246]: I think this Vertical Arrangement here might be why the parkings all display in a vertical line? I understand it's for symmetry with the Reviews but honestly it just looks a bit off on the screen :/. Would it be possible to put 3 Parkings per line?

**Review by LouisJamesVasseur — COMMENTED** [review]
Loving this feature! Such a good idea and will allow us to incorporate images better and overall is a great addition to the Profile Screen

Screenshot 2024-12-07 at 11 53 01

I think the way that Favorite Parkings are displayed right now could be improved. As I said in my inline comment, 3 Parkings per line would make this perfect. For now (like you can see in my attached screenshot, it doesn't look very polished

File: app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt [hunk 2]
```diff
@@ -183,49 +179,71 @@ fun ViewProfileScreen(
...
+  Column(
+      modifier = Modifier.fillMaxWidth().padding(top = 56.dp) // Adjust based on button height
+      ) {
+        TabRow(selectedTabIndex = selectedTabIndex, modifier = Modifier.testTag("TabRow")) {
+          tabs.forEachIndexed { index, title ->
+            Tab(
+                text = { Text(title) },
+                selected = selectedTabIndex == index,
+                onClick = { selectedTabIndex = index },
+                modifier = Modifier.testTag("Tab${title.replace(" ", "")}"))
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt:217]: Why is this needed? Can't we just adjust the tab names in the tests?

File: app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt [hunk 3]
```diff
@@ -183,49 +179,71 @@ fun ViewProfileScreen(
...
+          }
+        }
 
-  Text(
-      text = stringResource(R.string.view_profile_screen_favorite_parking_title),
-      style = MaterialTheme.typography.titleLarge,
-      modifier = Modifier.testTag("FavoriteParkingsTitle"))
+        when (selectedTabIndex) {
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt:221]: making it like this will make it much easier to incorporate new tabs later, well done!

File: app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt [hunk 4]
```diff
@@ -265,7 +283,7 @@ private fun FavoriteParkingCard(
               modifier =
                   Modifier.align(Alignment.Center).padding(8.dp).testTag("ParkingItem_$index"))
 
-          IconButton(
+          androidx.compose.material3.IconButton(
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt:286]: Why was this specific import added?

File: app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt (hunk 1 above)
Comment (Armetiis) [reply app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt:246]: I don't understand the remark. This is not intended to stay like this, described in the "Future Improvements" section of the PR description. The cards will be horizontally elongated, just like in the list screen.

File: app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt (hunk 2 above)
Comment (Armetiis) [reply app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt:217]: I'm not a big fan of Tags with whitespaces honestly. This is not very important. I'm just more comfortable with it. Do you think I should change it?

File: app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt (hunk 4 above)
Comment (Armetiis) [reply app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt:286]: I just swapped the imports. Previously the material 3 IconButton was imported and so when using our atom Icon Button we had to specify cyrcle.main...IconButton. I think for consistency it's better to import our own composables by default and specify "material3" when we need to use a more customizable version

Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt:286]: Fair enough, thanks for clarifying

File: app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt (hunk 1 above)
Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt:246]: Did you see the picture I posted with my review? I just thought it looked really odd to have tiny things when scrolling, and I wasn't super convinced it should stay like this even with you mentioning in the PR description that the Parking Cards would be elongated. If you're sure this will be dealt with, though, I'll trust you.

Comment (Armetiis) [reply app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt:246]: I mean, you would be suggesting I should rework their UI layout when a PR will be specifically dedicated to that. I purposedly didn't change them (trust me it urges me) to let that PR take care of it. They are the exact same, just in a Column Layout rather than Row to match the Tab layout

Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt:246]: Cool, knowing it's on the way is enough for me

**Review by LouisJamesVasseur — APPROVED** [review]
Great work as usual. Thanks for explaining and I'm looking forward to the new Parking cards!

**Review by sean-prz — APPROVED** [review]
Love the UI improvement, approved !
